src/server/game/Maps/Map.cpp

```diff
@@ -879,10 +879,15 @@ void Map::RemovePlayerFromMap(Player* player, bool remove)
 {
     sScriptMgr->OnPlayerLeaveMap(this, player);
 
+    player->getHostileRefManager().deleteReferences(); // multithreading crashfix
```

Comment:
Called in ResurrectPlayer so absolutely not needed. Will apply all suggestions if someone decides it's worth merging.